### PR TITLE
Closes #24 and closes #25 - Domain types and Module Transaction mapping into custom

### LINKED ISSUE #24: Transaction domain types
**Is your feature request related to a problem? Please describe.**
The clients currently deserialise the polymorphic response from the `/safe/{address}/all-transactions` into `Multisig`, `Module` and `Ethereum` transactions. Then map each of those types into `Transfer`, `Settings` or `Custom` transaction types. 

**Describe the solution you'd like**
Deserialization is already largely implemented. The domain types (`Transfer`, `Settings` and `Custom`) should be defined according to the needs of the apps.

**Additional context**
For reference, the domain transaction data layouts expected by Android are as follow:

```kotlin
sealed class Transaction {
    abstract val status: TransactionStatus

    // If status is Successful, Failed or Canceled, the confirmations can be null
    abstract val confirmations: Int?

    data class Custom(
        override val status: TransactionStatus,
        override val confirmations: Int?,
        val nonce: BigInteger?,
        val address: Solidity.Address,
        val dataSize: Long,
        val date: String?,
        val value: BigInteger
    ) : Transaction()

    data class SettingsChange(
        override val status: TransactionStatus,
        override val confirmations: Int?,
        val dataDecoded: DataDecodedDto,
        val date: String?,
        val nonce: BigInteger
    ) : Transaction()

    data class Transfer(
        override val status: TransactionStatus,
        override val confirmations: Int?,
        val recipient: Solidity.Address,
        val sender: Solidity.Address,
        val value: BigInteger,
        val date: String?,
        val tokenInfo: ServiceTokenInfo?,
        val nonce: BigInteger?
    ) : Transaction()
}

enum class TransactionStatus() {
    AwaitingConfirmations,
    AwaitingExecution,
    Cancelled,
    Failed,
    Success,
    Pending // Not supported yet as these correspond to the ones issued from the device
}

```

### LINKED ISSUE #25: Map ModuleTransaction into transaction domain types
**Is your feature request related to a problem? Please describe.**
Module transactions should be mapped (at least so far) only into `Custom` as defined in #24 

**Describe the solution you'd like**
There should be a mechanism to detect and map any `ModuleTransaction` into a `CustomTransaction` 

**Additional context**
For reference, the classes described in #24 will be used. For mapping logic the `TransactionRepository` implementation will be used and filling out of data will be done as needed.

### PR DESCRIPTION
**Pull request by jpalvarezl**
- Status calculation still missing (depends on #28)
Closes #25 (and maybe #24 too?)

### CODE REVIEW

File: src/models/converters/transactions.rs [hunk 1]
```diff
@@ -115,27 +123,48 @@ impl MultisigTransaction {
     }
 
     fn to_custom(&self) -> Custom {
-        Custom { to: self.safe }
+        Custom {
+            to: self.safe,
+            data_size: self.data.as_ref().unwrap_or(&String::new()).len().to_string(),
```
Comment (rmeissner) [inline src/models/converters/transactions.rs:128]: length of data = (length of string - 2)

File: src/models/service/transactions.rs [hunk 2]
```diff
@@ -3,15 +3,44 @@ use serde::Serialize;
...
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum TransactionStatus {
+    AwaitingConfirmations,
+    AwaitingExecution,
+    Cancelled,
+    Failed,
+    Success,
+    Pending, //is this really necessary as it's a state only reachable in devices, afaik?
```
Comment (rmeissner) [inline src/models/service/transactions.rs:24]: get rid of it

File: src/services/transactions.rs [hunk 3]
```diff
@@ -45,9 +45,14 @@ pub fn get_all_transactions(safe_address: &String) -> Result<Vec<ServiceTransact
...
-    let transactions: Vec<ServiceTransaction> = transactions.results.into_iter()
-        .flat_map(|transaction| transaction.to_service_transaction())
+    let backend_transactions: Page<TransactionDto> = serde_json::from_str(&body)?;
+    let service_transactions: Vec<ServiceTransaction> = backend_transactions.results.into_iter()
+        .flat_map(|transaction| transaction.to_service_transaction().unwrap())
         .collect();
-    Ok(transactions)
+    Ok(Page {
+        count: service_transactions.len(),
+        next: backend_transactions.next,
```
Comment (rmeissner) [inline src/services/transactions.rs:54]: https://github.com/rmeissner/safe-merc-service/blob/master/src/renderer/overview.rs#L44

Comment (jpalvarezl) [reply src/services/transactions.rs:54]: Will address in #32

**Pull request merged**